src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/host-form/host-form.component.html

```diff
@@ -80,7 +80,8 @@
           </div>
 
           <!-- Maintenance Mode -->
-          <div class="form-group row">
+          <div class="form-group row"
+               *ngIf="pageURL === 'hosts'">
```

Comment:
That's ok, but (personally) I think it's preferable that conditions express what's the underlying reason for doing something.

There are multiple ways:
* For example, in the cluster expansion workflow, this component will be child/grandchild of the cluster expansion component, so if you inject that component in the constructor you'll be able to check that condition which really express why this is hidden (`if parentIsClusterCreation then hide this`).
* Even better, you can extend the interface of the hosts page with a new argument: `hideMaintenance` or `deploymentStage`, etc...

IMHO all those checks (`if parentIsClusterCreation`, `if hideMaintenance`, `if deploymentStage`) better reflect the intention of this change.

Reply:
done! can you guys please check it again @pereman2 @epuertat